server/rest/src/test/java/org/infinispan/rest/resources/CacheResourceV2Test.java

```diff
@@ -464,6 +465,51 @@ public void testCacheV2Distribution() {
       }
    }
 
+   @Test
+   public void testCacheV2KeyDistribution() {
+      final String cacheName = "keyDistribution";
+      String cacheJson = "{ \"distributed-cache\" : { \"statistics\":true } }";
+      createCache(cacheJson, cacheName);
+
+      RestCacheClient cacheClient = client.cache(cacheName);
+
+      putStringValueInCache(cacheName, "key1", "data");
+      putStringValueInCache(cacheName, "key2", "data");
+
+      CompletionStage<RestResponse> response = cacheClient.distribution("key1");
+      assertThat(response).isOk();
+
+      try (RestResponse restResponse = join(response)) {
+         Json jsonNode = Json.read(restResponse.getBody());
+         assertTrue(jsonNode.at("contains_key").asBoolean());
+         assertTrue(jsonNode.at("owners").isArray());
+
+         List<Json> distribution = jsonNode.at("owners").asJsonList();
+         assertEquals(NUM_SERVERS, distribution.size());
+
+         Pattern pattern = Pattern.compile(this.getClass().getSimpleName() + "-Node[a-zA-Z]$");
+         for (Json node : distribution) {
+            assertEquals(node.at("node_addresses").asJsonList().size(), 1);
+            assertTrue(pattern.matcher(node.at("node_name").asString()).matches());
+            assertTrue(node.has("primary"));
+         }
+      } catch (Exception e) {
+         throw new TestException(e);
+      }
+
+      response = cacheClient.distribution("inexistent");
+      assertThat(response).isOk();
+
+      try (RestResponse restResponse = join(response)) {
+         Json jsonNode = Json.read(restResponse.getBody());
+         assertFalse(jsonNode.at("contains_key").asBoolean());
+         assertTrue(jsonNode.at("owners").isArray());
+         assertEquals(0, jsonNode.at("owners").asJsonList().size());
```

Comment:
So if the key isn't present we don't show the distribution info? Why not always show the distribution info?

Reply:
Since we show which members hold a key, I feel it may be misleading to display information about members for a key that does not exist.

Reply:
But the `contains_key` field specifically says if the key is present or not.

Reply:
We should return all members in such cases?

Reply:
It really depends upon the use case, but I would expect if I am retrieving the distribution info for the key it would tell me who all the owners are irrespective of the key presence. Do you know this is designed to be used?

Reply:
I guess that from a UX perspective for example in the console, if the key is not present we don't care about the key distribution if it existed, but probably there are other uses cases that i'm missing for that case so if we can have the information better to have it than ignore it IMO.